src/databricks/labs/remorph/deployment/dashboard.py

```diff
@@ -1,82 +1,107 @@
-from typing import Any
+from pathlib import Path
 
 from databricks.labs.blueprint.installation import Installation
 from databricks.labs.blueprint.installer import InstallState
 from databricks.sdk import WorkspaceClient
-from databricks.sdk.errors import DatabricksError
-from databricks.sdk.errors import InvalidParameterValue
 from databricks.sdk.retries import retried
-from databricks.sdk.service.dashboards import Dashboard
+from databricks.sdk.errors import InvalidParameterValue, NotFound, DatabricksError, ResourceAlreadyExists
+
+from databricks.sdk.service.dashboards import LifecycleState
+from databricks.labs.lsql.dashboards import DashboardMetadata, Dashboards
+
+from databricks.labs.remorph.config import ReconcileMetadataConfig
 
 logger = logging.getLogger(__name__)
 
 
 class DashboardDeployment:
     _UPLOAD_TIMEOUT = timedelta(seconds=30)
 
-    def __init__(self, ws: WorkspaceClient, installation: Installation, install_state: InstallState):
+    def __init__(
+        self,
+        ws: WorkspaceClient,
+        installation: Installation,
+        install_state: InstallState,
+    ):
         self._ws = ws
         self._installation = installation
         self._install_state = install_state
 
-    def deploy(self, name: str, dashboard_file: Traversable, parameters: dict[str, Any] | None = None):
-        logger.debug(f"Deploying dashboard {name} from {dashboard_file.name}")
-        dashboard_data = self._substitute_params(dashboard_file, parameters or {})
-        dashboard = self._update_or_create_dashboard(name, dashboard_data, dashboard_file)
-        logger.info(f"Dashboard deployed with dashboard_id {dashboard.dashboard_id}")
-        logger.info(f"Dashboard URL: {self._ws.config.host}/sql/dashboardsv3/{dashboard.dashboard_id}")
+    def _handle_existing_dashboard(self, dashboard_id: str, display_name: str, parent_path: str) -> str | None:
+        try:
+            dashboard = self._ws.lakeview.get(dashboard_id)
+
+            if dashboard.lifecycle_state is None:
+                raise NotFound(f"Dashboard life cycle state: {display_name} ({dashboard_id})")
+            if dashboard.lifecycle_state == LifecycleState.TRASHED:
+                logger.info(f"Recreating trashed dashboard: {display_name} ({dashboard_id})")
+                return None  # Recreate the dashboard if it is trashed (manually)
+
+        except (NotFound, InvalidParameterValue):
+            logger.info(f"Recovering invalid dashboard: {display_name} ({dashboard_id})")
+            try:
+                dashboard_path = f"{parent_path}/{display_name}.lvdash.json"
+                self._ws.workspace.delete(dashboard_path)  # Cannot recreate dashboard if file still exists
+                logger.debug(f"Deleted dangling dashboard {display_name} ({dashboard_id}): {dashboard_path}")
+            except NotFound:
+                pass
+            return None  # Recreate the dashboard if it's reference is corrupted (manually)
```

Comment:
don't return None, bubble up NotFound and handle it in the client code.

here we have some anti-pattern logic:
```
try: 
  if not ...: raise NotFound
  return None
except NotFound:
  try:
     if not ...: raise NotFound
  except NotFound:
     return None
```